### PR TITLE
server, acl: use peer capabilities provided by tailscale

### PR DESCRIPTION
**Pull request by danderson**
Removes all internal ACL file support, but keeps the _internal namespace reserved for future use.

Updates tailscale/corp#13375

### CODE REVIEW

**Comment by danderson** [comment]
Bah, I did a late rename right before sending the PR and it broke everything. Sorry, fixing.

**Comment by danderson** [comment]
Okay, ready now, sorry for the false start.

File: acl/acl.go [hunk 1]
```diff
@@ -54,141 +24,61 @@ const (
...
-type compiledRule struct {
-	from   map[string]bool
-	secret *regexp.Regexp
-	action map[Action]bool
+// Rule is an access control rule that permits some actions on some
+// secrets. Secrets can contain '*' wildcards, which match zero or
+// more characters.
+type Rule struct {
+	Action []Action `json:"action"`
+	Secret []string `json:"secret"`
```
Comment (creachadair) [inline acl/acl.go:45]: Something to think about for a future change, we could define a little wrapper `type Secret string` that has a `Match` method to do the glob matching.

**Review by creachadair — APPROVED** [review]
A bit hairy, but not actually scary. 🙂

File: acl/acl.go (hunk 1 above)
Comment (danderson) [reply acl/acl.go:45]: Good call, I'll make a quick followup.

**Pull request merged**